finetune/base_models/gpt/encoder.py

```diff
@@ -188,31 +188,36 @@ def decode(self, ids):
         return "".join([self.decoder.get(word_idx, '<unk>') for word_idx in ids]).replace("</w>", " ")
 
 
-def to_spacy_attn(attn, tokens, token_starts, token_ends):
+def to_spacy_attn(attn, tokens, token_starts, token_ends, normalize=True):
     to_combine = []
     spacy_attn = []
     spacy_token_starts = []
     spacy_token_ends = []
     spacy_start = None
     for token, prob, start, end in zip(tokens, attn, token_starts, token_ends):
         to_combine.append(prob)
-        if not spacy_start:
+        if spacy_start is None:
             spacy_start = start
         if token.endswith('</w>'):
-            spacy_attn.append(max(to_combine))
+            if normalize:
+                spacy_attn.append(np.max(to_combine, 0))
+            else:
+                spacy_attn.append(to_combine[-1])
             spacy_token_starts.append(spacy_start)
             spacy_token_ends.append(end)
             to_combine = []
             spacy_start = None
-    spacy_attn = spacy_attn/sum(spacy_attn)
+            
+    if normalize:
+        spacy_attn = spacy_attn/sum(spacy_attn)
     return {
         'attention_weights': spacy_attn,
         'token_starts': spacy_token_starts,
         'token_ends': spacy_token_ends
     }
 
 
-def finetune_to_indico_attention_weights(raw_texts, attn_weights, encoder):
+def finetune_to_indico_attention_weights(raw_texts, attn_weights, encoder, attention=True):
```

Comment:
change name?

Reply:
Probably not as this is in the public API??

Reply:
Only nominally, it's not documented and I'm sure no one uses it.